File: src/main/java/models/ProductsData.java

package models;

import java.util.List;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class ProductsData {

    @SerializedName("nodes")
    @Expose
    private List<Node> nodes = null;
    @SerializedName("products")
    @Expose
    private List<Product> products = null;

    public List<Node> getNodes() {
        return nodes;
    }

    public void setNodes(List<Node> nodes) {
        this.nodes = nodes;
    }

    public List<Product> getProducts() {
        return products;
    }

    public void setProducts(List<Product> products) {
        this.products = products;
    }

}

File: src/main/kotlin/models/ProductsData.kt
package models

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class ProductsData(@SerializedName("nodes")
                   @Expose
                   val nodes: List<Node>,

                   @SerializedName("products")
                   @Expose
                   val products: List<Product>)
